core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -147,10 +139,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * @param jars Collection of JARs to send to the cluster. These can be paths on the local file
    *             system or HDFS, HTTP, HTTPS, or FTP URLs.
    * @param environment Environment variables to set on worker nodes.
-   * @param preferredNodeLocationData used in YARN mode to select nodes to launch containers on.
-   * Can be generated using [[org.apache.spark.scheduler.InputFormatInfo.computePreferredLocations]]
-   * from a list of input files or InputFormats for the application.
+   * @param preferredNodeLocationData not used. Left for backward compatibility.
    */
+  @deprecated("Passing in preferred locations has no effect at all, see SPARK-10921", "1.6.0")
```

Comment:
I was going to say, hm, it may be funny to deprecate and then un-deprecate this if `preferredNodeLocationData` comes back. But then again, another constructor with this parameter was already deprecated. But then that one was a `@DeveloperApi`. All in all I think this change is OK though, on grounds of consistency. Un-deprecating isn't a big deal.